resource/resource.go

```diff
@@ -94,6 +93,17 @@ func AsObject(s *structpb.Struct, o runtime.Object, opts ...json.Options) error
 	return errors.Wrapf(json.Unmarshal(b, o, opts...), "cannot unmarshal JSON from %T into %T", s, o)
 }
 
+// AsObject gets the supplied Kubernetes object from the supplied struct.
+func AsObject(s *structpb.Struct, o runtime.Object) error {
+	return asObject(s, o)
+}
+
+// AsObjectStrict gets the supplied Kubernetes object from the supplied
+// struct erroring out in case of unknown fields.
+func AsObjectStrict(s *structpb.Struct, o runtime.Object) error {
+	return asObject(s, o, json.RejectUnknownMembers(true))
+}
```

Comment:
I'd kinda prefer the approach of having our own options (not `json`'s). If this is the only variant we think will exist it might be fine, but I wouldn't want to end up with 3-4 `AsObjectFoo` functions.

Reply:
I agree, but I can't come up with a good API, so feel free to open a new PR if you have something in mind already.